packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -0,0 +1,1216 @@
+    final double width = clampDouble(
+      clampDouble(childSize.width + layoutOffset.dx.abs(), lowerBound, upperBound),
+      0,
+      childSize.width,
+    );
+    // Vertical
+    lowerBound = 0;
+    upperBound = viewportDimension.height + cacheExtent;
+    final double height = clampDouble(
+      clampDouble(childSize.height + layoutOffset.dy.abs(), lowerBound, upperBound),
+      0,
+      childSize.height,
+    );
+    return Size(width, height);
+  }
+
+  /// The offset at which the given `child` should be painted.
+  ///
+  /// The returned offset is from the top left corner of the inside of the
+  /// viewport to the top left corner of the paint coordinate system of the
+  /// `child`.
+  ///
+  /// This is useful when the one or both of the axes of the viewport are
+  /// reversed. The normalized layout offset of the child is used to compute
+  /// the paint offset in relation to the [verticalAxisDirection] and
+  /// [horizontalAxisDirection].
+  @protected
+  Offset computeAbsolutePaintOffsetFor(
+    RenderBox child, {
+    required Offset layoutOffset,
+    required Size paintExtent,
+  }) {
+    assert(hasSize); // this is only usable once we have a size
+    final double xOffset;
+    final double yOffset;
+    switch (verticalAxisDirection) {
+      case AxisDirection.up:
+        yOffset = viewportDimension.height - (layoutOffset.dy + paintExtent.height);
+      case AxisDirection.down:
+        yOffset = layoutOffset.dy;
+      case AxisDirection.right:
+      case AxisDirection.left:
+        throw Exception('This should not happen');
+    }
+    switch(horizontalAxisDirection){
+      case AxisDirection.right:
+        xOffset = layoutOffset.dx;
+      case AxisDirection.left:
+        xOffset = viewportDimension.width - (layoutOffset.dx + paintExtent.width);
+      case AxisDirection.up:
+      case AxisDirection.down:
+        throw Exception('This should not happen');
+    }
+    return Offset(xOffset, yOffset);
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (_children.isEmpty) {
+      return;
+    }
+    if (_hasVisualOverflow && clipBehavior != Clip.none) {
+      _clipRectLayer.layer = context.pushClipRect(
+        needsCompositing,
+        offset,
+        Offset.zero & viewportDimension,
+        _paintChildren,
+        clipBehavior: clipBehavior,
+        oldLayer: _clipRectLayer.layer,
+      );
+    } else {
+      _clipRectLayer.layer = null;
+      _paintChildren(context, offset);
+    }
+  }
+
+  void _paintChildren(PaintingContext context, Offset offset) {
+    for (final RenderBox child in childrenInPaintOrder) {
+      if (parentDataOf(child).isVisible) {
+        context.paintChild(child, offset + parentDataOf(child).paintOffset);
+      }
+    }
+  }
+
+  // ---- Called from _TwoDimensionalViewportElement ----
+
+  void _insertChild(RenderBox child, ChildVicinity slot) {
+    if (slot.xIndex == 0 && slot.yIndex == 0) {
+    }
```

Comment:
this if is empty 😆

Reply:
(It's surprising that no lint is flagging this....)

Reply:
Oh my, I must have missed something here. I wonder if this is just a merge issue I had. I will fix! :)